File: publication-figure-renderer/python/src/publication_figure_renderer/output_writer.py
"""Writes an SVG document to a file on disk.

Isolating this keeps the rest of the pipeline testable without touching
the filesystem.
"""

from __future__ import annotations

from pathlib import Path


def write_svg(svg_text: str, path: str | Path) -> Path:
    """Write ``svg_text`` to ``path`` as UTF-8.  Returns the path for chaining."""
    target = Path(path)
    target.write_text(svg_text, encoding="utf-8")
    return target

File: publication-figure-renderer/r/R/output_writer.R
# Writes an SVG document to a file on disk.  Isolating this makes the
# rest of the pipeline testable without touching the filesystem.

#' Write an SVG document to disk as UTF-8.
#'
#' @param svg_text The SVG document as a single character string.
#' @param path File path to write.
#' @return `path` (invisibly), for chaining.
write_svg <- function(svg_text, path) {
  writeBin(charToRaw(enc2utf8(svg_text)), path)
  invisible(path)
}
